modules/nicknames/events/guildMemberUpdate.js

```diff
@@ -0,0 +1,59 @@
+const {truncate} = require("../../../src/functions/helpers");
+module.exports.run = async function (client, oldGuildMember, newGuildMember) {
+
+    if (!client.botReadyAt) return;
+    if (newGuildMember.guild.id !== client.guild.id) return;
+    if (newGuildMember.nickname === oldGuildMember.nickname && newGuildMember.roles.cache.size === oldGuildMember.guild.me.roles.cache.size) return;
+
+    const roles = client.configurations['nicknames']['config'];
+    const moduleModel = client.models['nicknames']['User'];
+
+    let rolePrefix = '';
+    let userRoles = newGuildMember.roles.cache.map(r => r.id);
```

Comment:
```suggestion
    let userRoles = newGuildMember.roles.cache.sort((a, b) => a.position - b.position).map(r => r.id);
```
Sorting by position ensures that the highest role gets added

Reply:
Now botReady works but it gives me this error here:

TypeError: Cannot read properties of undefined (reading 'cache')
    at renameMember (D:\dev\WebstormProjects\CustomDCBot\modules\nicknames\renameMember.js:8:39)
    at module.exports.run (D:\dev\WebstormProjects\CustomDCBot\modules\nicknames\events\botReady.js:6:15)
    at Client.<anonymous> (D:\dev\WebstormProjects\CustomDCBot\main.js:413:103)
    at Client.emit (node:events:518:28)
    at startUp (D:\dev\WebstormProjects\CustomDCBot\main.js:193:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Reply:
To investigate this, I'd need access to your WIP files, please either perform a commit and push or send me both files on discord. Thanks.